### ISSUE TITLE
Loops are disabled or fail when using beatjump to move a loop forward and backwards too quickly.

### ISSUE THREAD

**Issue by CaMeLCaseError** [mixxxdj/mixxx#12901]
### Bug Description

When rotary wheels are used to "beat jump" the active loops often fail after some time of moving around.

Expected: While a loop is activated the whole loop gets moved by beatjumps forward and backwards.
This expected behavior works when the rotary encoder is turned very gently. Turning the rotary encoder a little quicker frequently disables the loop and "normal" play continues. Sometimes the loop stays active (visibly the region is still marked, the loop button is also still active) but playing continues as if no loop was activated. 

I can reproduce the issue easily with 1/32 loops and 1/16 beat step. But also on long loops (4 beats for example) I was able to reproduce the issue. (i guess it's related to the frequency of hitting the loop borders).

It happens less when:
* loop size increases
* jump size stays small compared to loop size.
 
Controllers tested: hercules P32 official mapping and some of my own mappings.

I made a screen cast but didn't find a place to quickly upload. Any recommendations and I'll upload and link it here.

### Version

2.5-alpha-255-g2d2d5f304b (main)

### OS

windows, linux

**Comment by ronso0** [comment]
Oh.. Can you reproduce the issue with Mixxx 2.4?

**Comment by CaMeLCaseError** [comment]
Yes. I just spent some minutes doing the same in 2.4 (windows) and interestingly had the impression the issue happens less likely. But it still happens. (also still in both ways that: either the loop button stays visibly enabled or sometimes it's also getting disabled)

**Comment by ronso0** [comment]
Thanks for testing!

Can you confirm the issue happen less likely (in 2.5/main) if you increse your audio buffer siez?

**Comment by ronso0** [comment]
I can confirm it with Mixxx 2.4

**Comment by ronso0** [comment]
(actually a dupe of my own, old report #11261 but let's continue here)

**Comment by CaMeLCaseError** [comment]
> Thanks for testing!
> 
> Can you confirm the issue happen less likely (in 2.5/main) if you increse your audio buffer siez?
just downloaded fresh 2.5 from 5th of march.

If i lower the buffer size to 1.3ms (using 96khz samplerate) size the issue goes away completely. that's a big surprise...
just tested with 1/32 loops on 100bpm.

Also I noticed the loops sound much cleaner now with the low buffer size while moving around. Before when i was using 20ms there where clicks when moving the loop.

**Comment by ronso0** [comment]
Interesting, thanks for testing!

I still can't reproduce it reliably, regardless the buffer size, even though I expected it to happen more often with short buffers (more underruns).
I added some debug prints and will try again.

**Comment by ronso0** [comment]
@CaMeLCaseError In case you [run Mixxx from the command line](https://manual.mixxx.org/2.4/en/chapters/appendix/commandline_dev_tools.html), do you see this when the clicks occur?
`warning [Engine] ERROR: Couldn't get all needed samples for crossfade.`

**Comment by CaMeLCaseError** [comment]
didn't see any output when hearing these clicks. I'll try to come up with a test tone that is easy to debug. maybe a slowly raising sine wave that's easy to visually debug inside a wave editor.
